### ISSUE TITLE
Support for named tags in Trans components prop

### ISSUE THREAD

**Issue by AlexStansfield** [vinissimus/next-translate#225]
Relying on indexes for the components we feed into the Trans components prop is a bit of a pain and would be great if we could do something like this:

```
    <Trans
      i18nKey={i18nKey}
      components={{
        strong: <span style={{ fontWeight: 500 }} />,
        black: <span style={{ color: '#555555' }} />,
        green: <span style={{ color: '#21c794' }} />,
        red: <span style={{ color: '#e60012' }} />,
        br: <br />,
      }}
      values={values}
    />
```

with this in the locale:

```
   "repair_type": "<strong><green>{{value}}</green></strong> repair",
```

I see the regex in the code only matches for digits in the tag. Could it also match for any keys if an object is passed into the components prop?

**Comment by LuisFros** [comment]
Would be great! as  `next-i18next` does

**Comment by aralroca** [comment]
@AlexStansfield @LuisFros this will be available on 0.20! Already available on the prerelease [0.20.0-canary.3](https://github.com/vinissimus/next-translate/releases/tag/0.20.0-canary.3). If you try it, leave feedback to know that it works well for you, and if you see something wrong don't hesitate to report it 🙏  ! Thank you very much 😊
